include/s3select.h

```diff
@@ -2282,12 +2343,49 @@ class base_s3object
 	return m_returned_bytes_size;
   }
 
-  void result_values_to_string(multi_values& projections_resuls, std::string& result)
+  void json_result_format(multi_values& projections_results, std::string& result, std::string& output_delimiter)
   {
-    size_t i = 0;
+    std::string output_row_delimiter(1,m_csv_defintion.output_row_delimiter);
+    std::vector<std::string> key_path{};
+
+    result += "{";
+    int j = 0;
+    for (size_t i = 0; i < projections_results.values.size(); ++i)
+    {
+      auto& res = projections_results.values[i];
+      std::string label = "_";
+      label += std::to_string(i + 1);
+
+      if (i > 0) {
+        result += output_delimiter;
+      }
+      
+      if(!is_star && !col_ref)  {
+        result += "\"" + label + "\":";
+        } else if(!is_star && col_ref) {
+          result += "\"" + m_projection_keys[j] + "\":";
+        } 
```

Comment:
the `m_projection_keys` contain all labels.
so, why prepping it again in this method?